hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobFileCleanerChore.java

```diff
@@ -94,8 +94,9 @@ private void checkObsoleteConfigurations() {
     }
   }
 
-  public MobFileCleanerChore() {
+  public MobFileCleanerChore(Admin admin) {
```

Comment:
Where do we use this method? I think a chore should only be executed at master side?

Reply:
Typically yes, but this class is also used by MobStressToolRunner and IntegrationTestMobCompaction, which are both effectively HBase clients.

Reply:
I guess these classes just want to run the cleanup method, instead of running a background task? Let me check the code...

Reply:
OK, checked the code, they both only call the cleanupObsoleteMobFiles method. I think we could introduce something like MobCleanupUtil, and put the cleanupObsoleteMobFiles and archiveMobFiles in it, where we need to change them to static methods, and then call them from the two classes, and also here, in MobFileCleanerChore. In this way we do not need to maintain this constructor then.